File: JavaScript/Deutschstunde.js
function derDieDas(wort){
  const arrVowelsUmlaute = ['a', 'e', 'i', 'o', 'u', 'ä', 'ö', 'ü'];
  const countWort = wort
    .toLowerCase()
    .split("")
    .filter((letter) => arrVowelsUmlaute
      .includes(letter));
  
  if (countWort.length < 2) return `das ${wort}`;
  else if (countWort.length === 2 || countWort.length === 3) return `die ${wort}`;
  else return `der ${wort}`;
}

console.assert(derDieDas('OOOpa') === 'der OOOpa', 'erro no test 01')
console.assert(derDieDas('Suppenhuhn') === 'die Suppenhuhn', 'erro no test 02')
console.assert(derDieDas('Geheimdienst') === 'der Geheimdienst', 'erro no test 03')
console.assert(derDieDas('Ofenkartoffel') === 'der Ofenkartoffel', 'erro no test 04')
console.assert(derDieDas('Fisch') === 'das Fisch', 'erro no test 05')
console.assert(derDieDas('Knödel') === 'die Knödel', 'erro no test 06')
console.assert(derDieDas('Leberkaassemmi') === 'der Leberkaassemmi', 'erro no test 07')

// ou

function derDieDas(wort) {
  var n = wort.split(/[aeiouäöü]/i).length - 1
  if (n < 2) return "das " + wort
  if (n > 3) return "der " + wort
  return "die " + wort
}

File: Python/deutschstunde.py
def der_die_das(wort):
    list_volwels_umlaute = 'aeiouäöü'
    count_wort = 0

    for letter in wort:
        if letter.lower() in list_volwels_umlaute:
            count_wort += 1

    if count_wort < 2:
        return f"das {wort}"
    elif count_wort == 2 or count_wort == 3:
        return f"die {wort}"
    else:
        return f"der {wort}"


print(der_die_das('OOOpa'))
print(der_die_das('Suppenhuhn'))
print(der_die_das('Geheimdienst'))
print(der_die_das('Ofenkartoffel'))
print(der_die_das('Fisch'))
print(der_die_das('Knödel'))
print(der_die_das('Leberkaassemmi'))
